src/nyx/features/ngldm.cpp

```diff
@@ -0,0 +1,346 @@
+			//		where
+			//	\mu = \sum_{i=1}^{N_g} \sum_{j=1}^{N_n} j\,p_{ij}
+			f_DCV += (k - f_DCM) * (k - f_DCM) * pij;	
+		}
+	}
+	f_LDE /= Ns;	// Low Dependence Emphasis	F_{\mathit{ngl.LDE}} = \frac{1}{N_s} \sum_{j=1}^{N_n} \frac{s_{.j}}{j^2}
+	f_HDE /= Ns;	// High Dependence Emphasis	F_{\mathit{ngl.HDE}} = \frac{1}{N_s} \sum_{j=1}^{N_n} j^2 s_{.j}
+	f_LGLCE /= Ns;	// Low Grey Level CountEmphasis	F_{\mathit{ngl.LGLCE}}=\frac{1}{N_s} \sum_{i=1}^{N_g} \frac{s_{i.}}{i^2}
+	f_HGLCE /= Ns;	// High Grey Level CountEmphasis	F_{\mathit{ngl.HGLCE}}=\frac{1}{N_s} \sum_{i=1}^{N_g} i^2 s_{i.}
+	f_LDLGLE /= Ns;	// Low Dependence Low Grey Level Emphasis	F_{\mathit{ngl.LDLGE}}=\frac{1}{N_s} \sum_{i=1}^{N_g} \sum_{j=1}^{N_n} \frac{s_{ij}}{i^2 j^2}
+	f_LDHGLE /= Ns;	// Low Dependence High Grey Level Emphasis	F_{\mathit{ngl.LDHGE}}=\frac{1}{N_s} \sum_{i=1}^{N_g} \sum_{j=1}^{N_n} \frac{i^2 s_{ij}}{j^2}
+	f_HDLGLE /= Ns;	// High Dependence Low Grey Level Emphasis	F_{\mathit{ngl.HDLGE}}=\frac{1}{N_s} \sum_{i=1}^{N_g} \sum_{j=1}^{N_n} \frac{j^2 s_{ij}}{i^2}
+	f_HDHGLE /= Ns;	// High Dependence High Grey Level Emphasis	F_{\mathit{ngl.HDHGE}}=\frac{1}{N_s} \sum_{i=1}^{N_g} \sum_{j=1}^{N_n} i^2 j^2 s_{ij}
+	f_GLNU /= Ns;	// Grey Level Non Uniformity	F_{\mathit{ngl.GLNU}}= \frac{1}{N_s} \sum_{i=1}^{N_g} s_{i.}^2
+	f_GLNUN /= (Ns * Ns);	// Grey Level Non Uniformity Normalised	F_{\mathit{ngl.GLNU.NORM}}= \frac{1}{N_s^2} \sum_{i=1}^{N_g} s_{i.}^2
+	f_DCNU /= Ns;	// Dependence Count Non Uniformity	F_{\mathit{ngl.DCNU}}= \frac{1}{N_s} \sum_{j=1}^{N_n} s_{.j}^2
+	f_DCNUN /= (Ns * Ns);	// Dependence Count Non Uniformity Normalised	F_{\mathit{ngl.DCNU.NORM}}= \frac{1}{N_s^2} \sum_{i=1}^{N_n} s_{.j}^2
+}
+
+void NGLDMfeature::save_value (std::vector<std::vector<double>>& fvals)
+{
+	fvals[NGLDM_LDE][0] = f_LDE;
+	fvals[NGLDM_HDE][0] = f_HDE;
+	fvals[NGLDM_LGLCE][0] = f_LGLCE;
+	fvals[NGLDM_HGLCE][0] = f_HGLCE;
+	fvals[NGLDM_LDLGLE][0] = f_LDLGLE;
+	fvals[NGLDM_LDHGLE][0] = f_LDHGLE;
+	fvals[NGLDM_HDLGLE][0] = f_HDLGLE;
+	fvals[NGLDM_HDHGLE][0] = f_HDHGLE;
+	fvals[NGLDM_GLNU][0] = f_GLNU;
+	fvals[NGLDM_GLNUN][0] = f_GLNUN;
+	fvals[NGLDM_DCNU][0] = f_DCNU;
+	fvals[NGLDM_DCNUN][0] = f_DCNUN;
+	fvals[NGLDM_GLM][0] = f_GLCM;
+	fvals[NGLDM_GLV][0] = f_GLV;
+	fvals[NGLDM_DCM][0] = f_DCM;
+	fvals[NGLDM_DCV][0] = f_DCV;
+	fvals[NGLDM_DCE][0] = f_DCE;
+	fvals[NGLDM_DCENE][0] = f_DCENE;
+}
+
+void NGLDMfeature::parallel_process_1_batch(size_t start, size_t end, std::vector<int>* ptrLabels, std::unordered_map <int, LR>* ptrLabelData)
+{
+	// Iterate ROIs of this batch
+	for (auto i = start; i < end; i++)
+	{
+		// Get ahold of ROI's cached data
+		int lab = (*ptrLabels)[i];
+		LR& r = (*ptrLabelData)[lab];
+
+		// Calculate feature of this ROI
+		NGLDMfeature f;
+		f.calculate(r);
+		f.save_value(r.fvals);
+	}
+}
+
+void NGLDMfeature::osized_add_online_pixel(size_t x, size_t y, uint32_t intensity) {}
+
+void NGLDMfeature::osized_calculate(LR& r, ImageLoader&)

```

Comment:
Is `ImageLoader&` used here?

Reply:
No, therefore its parameter is unnamed. May be modified in other children of class "FeatureMethod".